src/openforms/utils/json_logic.py

```diff
@@ -49,3 +52,174 @@ def normalize_value(value: Union[dict, list, tuple, int, str]) -> JSONLogicValue
         return JsonLogicTest.from_expression(value)
 
     raise NotImplementedError(f"Unknown value type: {type(value)}")
+
+
+@dataclass
+class ComponentMeta:
+    form_step: "FormStep"
+    component: Dict[str, Any]
+
+
+@dataclass
+class IntrospectionResult:
+
+    expression: dict
+    rule_tree: Dict[Any, str]
```

Comment:
```suggestion
    rule_tree: Dict[str, Any]
```
the dict keys are strings, the values are anything.